### PR TITLE
Validating provider description for urls in provider list view

### PR DESCRIPTION
**Pull request by amoghrajesh**
Validating URLs in Provider Descriptions for Provider List View

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by Lee-W — APPROVED** [review]
LGTM. should we add a test to it?

File: airflow/www/views.py [hunk 1]
```diff
@@ -4514,6 +4514,13 @@ def _clean_description(self, description):
         def _build_link(match_obj):
             text = match_obj.group(1)
             url = match_obj.group(2)
+
+            # parsing the url to check if ita a valid url
+            parsed_url = urlparse(url)
+            if not (parsed_url.scheme == "http" or parsed_url.scheme == "https"):
+                # returning the original raw text
+                return escape(match_obj.group(0))
+
```
Comment (hussein-awala) [inline airflow/www/views.py:4523]: Could you add a test for this change?
It is not just a validation, where the returned value would be different in some cases.

Comment (amoghrajesh) [reply airflow/www/views.py:4523]: Yeah, valid ask. I just pushed test cases.

**Comment by amoghrajesh** [comment]
> LGTM. should we add a test to it?

Yes it makes sense to add a test for this, @Lee-W can you take a look?

**Review by Lee-W — APPROVED** [review]
LGTM

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**